src/frontend/magnify/src/components/rooms/RegisterRoomForm/RegisterRoomForm.tsx

```diff
@@ -40,20 +42,29 @@ interface RegisterRoomFormValues {
   name: string;
 }
 
+interface FormErrors {
+  slug?: string[];
+}
+
 const RegisterRoomForm = ({ onSuccess }: RegisterRoomFormProps) => {
   const intl = useIntl();
   const validationSchema = Yup.object().shape({ name: Yup.string().required() });
   const queryClient = useQueryClient();
-  const { mutate, isLoading } = useMutation(RoomsRepository.create, {
-    onSuccess: (newRoom) => {
-      queryClient.setQueryData([MagnifyQueryKeys.ROOMS], (rooms: Room[] = []) => {
-        const newRooms = [...rooms];
-        newRooms.push(newRoom);
-        return newRooms;
-      });
-      onSuccess(newRoom);
+  const mutation = useMutation<Room, AxiosError, RegisterRoomFormValues>(
+    async (data: CreateRoomData) => {
+      return await RoomsRepository.create(data);
```

Comment:
```suggestion
RoomsRepository.create
```
https://eslint.org/docs/latest/rules/no-return-await